### ISSUE TITLE
New `erl_files_strict` (-hrl +beam) rule: `{elvis_style, code_complexity}`

### ISSUE THREAD

**Issue by elbrujohalcon** [inaka/elvis_core#285]
Inspired by https://hexdocs.pm/credo/Credo.Check.Refactor.ABCSize.html

Read the comments below to find more possible rules :)

**Comment by paulo-ferraz-oliveira** [comment]
There's also https://hexdocs.pm/credo/Credo.Check.Refactor.CyclomaticComplexity.html in a similar spirit.

**Comment by paulo-ferraz-oliveira** [comment]
And https://hexdocs.pm/credo/Credo.Check.Refactor.PerceivedComplexity.html.

**Comment by maco** [comment]
The assignments (A) part probably needs to be weighted differently for Erlang versus other languages. For example, I do WAY fewer assignments in Elixir than in Erlang, because in Elixir, there are pipes, and in Erlang there's a bunch of intermediate variables.

**Comment by paulo-ferraz-oliveira** [comment]
Agreed. This could probably be released initially as `experimental` (something I'm not sure we've ever done in `elvis_core`) and then tweaked with community feedback.

**Comment by elbrujohalcon** [comment]
Moving this one to **eventually 🙄** due to the complexity in its implementation.
If anybody feels brave enough to pick it up, by all means submit a PR and we can include it in the next release.

**Comment by NelsonVides** [comment]
These are too controversial to be defaults, adding them to the strict ruleset.
